### ISSUE TITLE
Create SortDescriptor From NSSortDescriptor

### ISSUE THREAD

**Issue by bdkjones** [realm/realm-cocoa#7412]
### Problem

`NSTableView` and `NSOutlineView` use `NSSortDescriptor` to allow sorting the table's contents. (This is what powers the "click on a column header to sort by that column" feature)

Realm does not use `NSSortDescriptor`. Realm offers its own `SortDescriptor`, which you can use to sort a `List<T>`, `Results<T>`, etc.

If the `NSSortDescriptor` uses a single-level key, such as a property of the exact objects powering the tableView, you can do this:

```
    let tableDescriptor: NSSortDescriptor = someTableView.sortDescriptors.first!
    let realmDescriptor: RealmSwift.SortDescriptor = RealmSwift.SortDescriptor(keyPath: tableDescriptor.key, ascending: tableDescriptor.ascending)
    // Go sort the Results<T> or List<T> powering the table with 'realmDescriptor', then reload the table.
```

However, if the `NSSortDescriptor` powering the tableView is using a **keypath** instead of a single level key, the above approach will not work. The `keypath` property of `NSSortDescriptor` is an `AnyKeyPath` type, which is not convertible to `String` (what Realm requires in its initializer).

It would be ludicrously useful if I could tell Realm: "Here's an `NSSortDescriptor`. Make a `SortDescriptor` out of it and if you can't, tell me about it." I understand that you've got your own custom SortDescriptor type because you don't support all the functionality that `NSSortDescriptor` has. But the constant back-and-forth between the UI and Realm's types needs to be easier here. 

Thank you.

### Solution

I'd like this API:

    RealmSwift.SortDescriptor.init(NSSortDescriptor: someNSSortDescriptor)

### How important is this improvement for you?

Would be a major improvement

**Comment by dianaafanador3** [comment]
HI @bdkjones have you been able to make this work in a simpler way?, Is the extension initialiser helpful for your use case?

**Comment by bdkjones** [comment]
@dianaafanador3 Nope. Because of #7413, I can't sort Realm's data the way I need to sort it anyway, so this issue is a moot point right now.
